lib/components/molecules/Typeahead/Typeahead.js

```diff
@@ -0,0 +1,172 @@
+  getSelectedSuggestionValue,
+}: Props): React.Element<any> => {
+  const [updatedSuggestions, setFilteredSuggestions] = useState(suggestions);
+  const [activeSuggestion, setActiveSuggestion] = useState(0);
+  const [showSuggestions, setShowSuggestions] = useState(true);
+
+  const triggerGetSuggestions = val => {
+    return val && val.trim().length >= 3;
+  };
+
+  /**
+   * Function to handle item selection
+   * @param {'*'} e
+   */
+  const handleClick = item => {
+    setActiveSuggestion(0);
+    getSelectedSuggestionValue(item);
+    setShowSuggestions(false);
+  };
+
+  const getFilteredSuggestions = (allSuggestions, inputText) => {
+    const filteredSuggestions = allSuggestions.filter(
+      suggestion => suggestion.toLowerCase().indexOf(inputText.toLowerCase()) > -1
+    );
+    return filteredSuggestions;
+  };
+
+  /**
+   * Function to handle uparrow, downarrow, enter and escape
+   * @param {*} e
+   */
+  const handleKeyDown = e => {
+    switch (e.keyCode) {
+      case ENTER_KEY: {
+        setActiveSuggestion(0);
+        getSelectedSuggestionValue(updatedSuggestions[activeSuggestion]);
+        setShowSuggestions(false);
+        break;
+      }
+      case UP_ARROW: {
+        if (activeSuggestion === 0) {
+          setActiveSuggestion(updatedSuggestions.length - 1);
+          return;
+        }
+        setActiveSuggestion(activeSuggestion - 1);
+        break;
+      }
+      case DOWN_ARROW: {
+        if (activeSuggestion + 1 === updatedSuggestions.length) {
+          setActiveSuggestion(0);
+          return;
+        }
+        setActiveSuggestion(activeSuggestion + 1);
+        break;
+      }
+      case ESCAPE: {
+        setActiveSuggestion(0);
+        setShowSuggestions(false);
+        break;
+      }
+      default:
+        break;
+    }
+  };
+
+  /**
+   * Function to handle text change in input
+   * @param {*} e
+   */
+  const handleChange = e => {
+    if (onChangeHandler) {
+      onChangeHandler(e);
+    }
+    // Note: Need to handle Debounce
+    const { value: inputText } = e.currentTarget;
+    if (triggerGetSuggestions(inputText)) {
+      if (getSuggestions) {
+        getSuggestions();
+        setFilteredSuggestions(suggestions);
+      } else {
+        const filteredSuggestions = getFilteredSuggestions(suggestions, inputText);
+        setFilteredSuggestions(filteredSuggestions);
+      }
+      setShowSuggestions(true);
+    } else {
+      setShowSuggestions(false);
+    }
+  };
+
+  const listOptions = {
+    inline: false,
+    nested: false,
+    noMargin: true,
+    inheritedStyles: '',
+    borderSeparator: false,
+    noDefaultView: false,
+  };
+
+  return (
+    <div className={className}>
+      <Input
+        id={id}
+        name={name}
+        type="search"
+        disabled={disabled}
+        value={value}
+        onChange={handleChange}
+        onKeyDown={handleKeyDown}
+        label=""
+        placeholder={placeholder}
+        autoComplete="off"
```

Comment:
How will a disabled use know if this is a type ahead suggestion box?

https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html
https://accessibility.huit.harvard.edu/%E2%9C%8E-technique-aria-autocomplete
https://uxmastery.com/anatomy-of-an-accessible-auto-suggest/